mongodbatlas/fw_data_source_mongodbatlas_project.go

```diff
@@ -0,0 +1,261 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20230201002/admin"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ datasource.DataSource = &ProjectDS{}
+var _ datasource.DataSourceWithConfigure = &ProjectDS{}
+
+var _ datasource.DataSource = &ProjectDS{}
+var _ datasource.DataSourceWithConfigure = &ProjectDS{}
+
+func NewProjectDS() datasource.DataSource {
+	return &ProjectDS{}
+}
+
+type ProjectDS struct {
+	client *MongoDBClient
+}
+
+type tfProjectDSModel struct {
```

Comment:
Any reason why this can not be projectModel?
```suggestion
type projectModel struct {
```

Reply:
That was to avoid confusion with atlas models, any models/types that might be created only for tests or temporarily otherwise for any computation. I initally did have `projectModel` but overall I found `tf` prefix to be much more readable especially when dealing with conversion functions, etc, that this is specific to the actual terraform plan/state/config model in question.

Reply:
As mentioned [here](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1366#discussion_r1284562820), I have found the `tf` prefix helpful when reading and developing.

Reply:
Discussed this with @andreaangiolillo offline as well. We will continue using `tf` an `fw` naming conventions. 
This will also be captured in this migration wiki as well.